src/app/GitExtensions.Extensibility/Git/GitItemStatus.cs

```diff
@@ -48,6 +48,9 @@ public GitItemStatus(string name)
     {
         Requires.NotNull(name, nameof(name));
         Name = name;
+
+        int pos = Name.LastIndexOf('/');
+        Path = RelativePath.From(pos >= 1 ? Name[..pos] : "");
```

Comment:
Note: This is only used FileStatusSorter.cs. This requires some memory, I would expect garbage collection is the major problem.
This could be cached in FileStatusSorter.cs only, then a slice could be used.
Really, just the position need to be stored.
Adds complexity, this could be a Lazy field for the situations not requiring Path

Reply:
The `Path` property is always used - unconditionally:

![image](https://github.com/user-attachments/assets/db98efd6-ce52-445f-a47b-8789b5b47a9a)

Reply:
Only when StatusSorter is used